src/geff/interops/trackmate_xml.py

```diff
@@ -875,7 +877,7 @@ def _build_geff_metadata(
         node_props_metadata=props_metadata["node_props_metadata"],
         edge_props_metadata=props_metadata["edge_props_metadata"],
         track_node_props={"lineage": "TRACK_ID"},
-        related_objects=[{"type": "image", "path": img_path}] if img_path is not None else None,
+        related_objects=[RelatedObject(type="image", path=img_path)] if img_path else None,
```

Comment:
one more comment here. avoid leaning on pydantic's ability to cast dictionaries to model instances for internal code.  Prefer using the actual objects (it makes refactoring much easier)